src/bsoncxx/decimal128.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2014 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include <bsoncxx/stdx/string_view.hpp>
+
+#include <bsoncxx/config/prelude.hpp>
+
+namespace bsoncxx {
+BSONCXX_INLINE_NAMESPACE_BEGIN
+
+///
+/// Represents an IEEE 754-2008 Decimal128 value in a platform-independent way.
+///
+class BSONCXX_API decimal128 {
+   public:
+    ///
+    /// Constructs a Decimal128 value representing zero.
+    ///
+    decimal128() : _high(0), _low(0){};
+
+    ///
+    /// Constructs a Decimal128 from high and low 64-bit big-endian parts.
+    ///
+    /// @param high
+    ///   High bits
+    /// @param low
+    ///   Low bits
+    ///
+    decimal128(const uint64_t h, const uint64_t l) noexcept : _high(h), _low(l){};
```

Comment:
Nixed const.  Here, I expanded the variable names from single-letter because it leads to better documentation with doxygen.  Elsewhere, I'm a firm believer in [short variables in small scopes]{https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-name-length}.

The idea behind noexcept here is that it can't throw an exception.  That's different than initialization from string, where the string might be bad.  I think the OID ones lacking noexcept is either because they are documented as possibly throwing or else an omission.